basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/UndertowOperations.java

```diff
@@ -31,106 +16,55 @@
-        String name = RandomStringUtils.randomAlphanumeric(6);
-        client.apply(new AddUndertowListener.HttpBuilder(name, httpServer, createSocketBinding()).enabled(true).build());
-        administration.reloadIfRequired();
-        return name;
     }
 
-    public void removeHTTPListener(String httpServer, String listenerName) throws CommandFailedException, InterruptedException, IOException, TimeoutException {
-        removeListener(httpServer, listenerName, UndertowListenerType.HTTP_LISTENER);
-    }
-
-    public String createHTTPSListener(String httpServer) throws IOException, CommandFailedException, TimeoutException, InterruptedException, OperationException {
-        String name = RandomStringUtils.randomAlphanumeric(6);
-        String realmName = createSecurityRealm();
-        client.apply(new AddUndertowListener.HttpsBuilder(name, httpServer, createSocketBinding())
-                .securityRealm(realmName)
-                .verifyClient(SslVerifyClient.NOT_REQUESTED)
-                .enabled(true)
+    /**
+     * @return name of created security realm
+     */
+    public String createSecurityRealm(String name) throws CommandFailedException, InterruptedException, TimeoutException, IOException {
+        client.apply(new AddHttpsSecurityRealm.Builder(name)
+                .keystorePassword("random")
+                .truststorePassword("random")
+                .keyPassword("random")
+                .keystorePath(getClass().getClassLoader().getResource("clientkeystore").getPath())
                 .build());
-        administration.reloadIfRequired();
+        administration.reload();
         return name;
     }
 
-    public void removeHTTPSListener(String httpServer, String listenerName) throws CommandFailedException, InterruptedException, IOException, TimeoutException {
-        removeListener(httpServer, listenerName, UndertowListenerType.HTTPS_LISTENER);
-    }
-
     /**
-     * @return name of created security realm
+     * If keepReference is set to true, name of created security realm name will be kept internally in UndertowOperations
+     * @param keepReference whether the name of newly created security realm will be saved
      */
-    public String createSecurityRealm() throws CommandFailedException, IOException,
-            InterruptedException, TimeoutException {
-        String realmName = "SecurityRealm_" + RandomStringUtils.randomAlphanumeric(6);
-        client.apply(new AddHttpsSecurityRealm.Builder(realmName)
-            .keystorePassword("random")
-            .truststorePassword("random")
-            .keyPassword("random")
-            .keystorePath(getClass().getClassLoader().getResource("clientkeystore").getPath())
-            .build());
-        administration.reload();
-        return realmName;
+    public String createSecurityRealm(boolean keepReference) throws InterruptedException, CommandFailedException, TimeoutException, IOException {
```

Comment:
Is `keepReference` argument necessary? Dou you really expect false value to be passed?